src/utility-types.ts

```diff
@@ -1,6 +1,12 @@
 export type Draft<T> = { -readonly [P in keyof T]?: T[P] }
 
-export interface MessageValue<Data> {
+export type JSONPrimitive = number | boolean | string | null
```

Comment:
we are sure that there are no other JSON primitives that we want to list here?

Reply:
Which are you thinking of?

Reply:
Nothing specific but you know types can be more and more and if we miss any here we will cause bugs.
This is my concern

Reply:
There are only a limited number of primitive types
The latest (`bigint`) is so new that few scientists use it for big calculations

https://developer.mozilla.org/en-US/docs/Glossary/Primitive

> There are 7 primitive data types: string, number, bigint, boolean, undefined, symbol, and null.

- `string`, `number`, `boolean` and `null` are used
- `undefined` is covered by the indexed object `JSONObject`
- `symbol` cannot be serialized as JSON values
- `bigint` could be added but also this is _currently_ [not supported](https://golb.hplar.ch/2019/01/js-bigint-json.html) by JSON :warning:  

... so indeed we have every JSON primitiv covered

Reply:
mmmmmmmmmmmmmmm @jerome-benoit  what do you think about this?
@Shinigami92 I think is really useful to define this and document it in JSDoc, but we should not apply that to our interface but on the concrete implementations, this is because we don't know if in future some other pool will support any type

Reply:
That indeed makes sense! So I will remove it from the abstract Pool and Worker.  
I will try if `unknown` can be applied for the abstraction, so we don't use the evil `any` type

Reply:
Great, I guess we can proceed I think this is beneficial for users

Reply:
> That indeed makes sense! So I will remove it from the abstract Pool and Worker.
> I will try if `unknown` can be applied for the abstraction, so we don't use the evil `any` type
> 
> Edit: done

You can also use Record<string, unknown> to ensure a key/value structure is required.